packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -672,13 +748,251 @@ class ZoomPageTransitionsBuilder extends PageTransitionsBuilder {
     return _ZoomPageTransition(
       animation: animation,
       secondaryAnimation: secondaryAnimation,
-      allowSnapshotting: allowSnapshotting && (route?.allowSnapshotting ?? true),
+      allowSnapshotting: allowSnapshotting && route.allowSnapshotting,
       allowEnterRouteSnapshotting: allowEnterRouteSnapshotting,
       child: child,
     );
   }
 }
 
+typedef AndroidBackGestureTransitionBuilder<T> = Widget Function(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+    AndroidBackEvent? startBackEvent,
+    AndroidBackEvent? currentBackEvent);
+
+class AndroidBackGesturePageTransitionsBuilder extends PageTransitionsBuilder {
+  const AndroidBackGesturePageTransitionsBuilder({
+    this.parent = const ZoomPageTransitionsBuilder(),
```

Comment:
If I just press the back button on a native Android predictive back route, it doesn't look like the zoom transition. It looks like a sped up version of the predictive back animation, I think.

[Screen_recording_20240220_154728.webm](https://github.com/flutter/flutter/assets/389558/d827422f-f6a4-4042-86c9-1161f6db527e)